SharedPackages/BrowserServicesKit/Sources/Subscription/Managers/SubscriptionManagerV2.swift

```diff
@@ -239,17 +239,20 @@ public final class DefaultSubscriptionManagerV2: SubscriptionManagerV2 {
 
     func migrateAuthV1toAuthV2IfNeeded() async {
 
-        guard v1MigrationNeeded else {
+        guard v1MigrationNeeded, // stops multiple attempts in a session, even in case of unrecoverable failures
+              !isUserAuthenticated else { // warning: using `isUserAuthenticated()` would create a loop
```

Comment:
This line is counter intuitive, since following the logic one would expect to actually do the migration when the user is authenticated. I get the meaning and suggest replacing it with an explicit check if `oAuthClient.currentTokenContainer` is not present

Reply:
This makes sense, changed